components/notion/notion.app.js

```diff
@@ -0,0 +1,586 @@
+const axios = require("axios");
+const get = require("lodash/get");
+const retry = require("async-retry");
+
+module.exports = {
+  type: "app",
+  app: "notion",
+  propDefinitions: {
+    blockId: {
+      type: "string",
+      label: "Page or Block Id",
+      description: "Unique identifier of a page or block. A block is an object that represents content within Notion, such as text, lists, media, etc. Even a page is a type of block, too.",
+    },
+    databaseId: {
+      type: "string",
+      label: "Database Id",
+      description: "Unique identifier of the database.",
+      async options() {
+        const options = [];
+        const notionDatabases = await this.getAllDatabases();
+        notionDatabases.forEach((database) => {
+          const title = this._getDatabaseTitle(database);
+          options.push({
+            label: title ?
+              title :
+              "untitled database",
+            value: database.id,
+          });
+        });
+        return options;
+      },
+    },
+    pageId: {
+      type: "string",
+      label: "Page Id",
+      description: "Unique identifier of the page.",
+      async options() {
+        const options = [];
+        const pages = await this.getAllItems("page");
+        pages.forEach( (page) =>  {
+          if ([
+            "page",
+          ].includes(page.object)) {
+            const title = this._getPageTitle(page);
+            options.push({
+              label: title ?
+                title :
+                page.id,
+              value: page.id,
+            });
+          }
+        });
+        return options;
+      },
+      optional: true,
+    },
+    startCursor: {
+      type: "string",
+      label: "Start Cursor",
+      description:
+        "If supplied, a page of results starting after the cursor provided will be returned. Otherwise, the first page of results will be returned.",
+      optional: true,
+    },
+    pageSize: {
+      type: "integer",
+      label: "Page Size",
+      description:
+        "The number of items from the full list desired in the response. Maximum: 100",
+      max: 100,
+      optional: true,
+    },
+    properties: {
+      type: "object",
+      label: "Properties",
+      description:
+        "An object with property values of the database or page. The keys are the names or IDs of the [property](https://developers.notion.com/reference-link/database-property) and the values are [property values](https://developers.notion.com/reference-link/page-property-value).",
+    },
+  },
+  methods: {
+    _authToken() {
+      return this.$auth.oauth_access_token;
+    },
+    _apiUrl() {
+      return "https://api.notion.com";
+    },
+    async _makeRequest(opts) {
```

Comment:
We need to use the [Notion SDK](https://github.com/makenotion/notion-sdk-js) instead.